torch/csrc/autograd/variable.h

```diff
@@ -384,13 +388,26 @@ struct TORCH_API DifferentiableViewMeta : public AutogradMeta {
   /// version_counter.current_version().
   uint32_t attr_version;
 
+  /// By default we use as_strided to recover views which is more efficient.
+  /// view_fn is only saved when as_strided is not supported.
+  /// If view_fn has value, we use it to recover views in backward.
+  c10::optional<std::function<at::Tensor(const at::Tensor&)>> view_fn_;
+
   CreationMeta creation_meta;
 
   bool requires_grad() const override {
     return requires_grad_ || grad_fn_ || (is_view_ && base_.requires_grad());
   }
 
-  DifferentiableViewMeta(at::TensorImpl* self_impl, Variable base,
+  bool has_view_fn() const {
+    return view_fn_.has_value();
+  }
+
+  std::function<at::Tensor(const at::Tensor&)> view_fn() const {
+    return view_fn_.value();
+  }
```

Comment:
It would be nice if this explicitly mentioned that it could fail if there is no view function

Reply:
Nice catch! Fixed :D